crates/transaction-pool/src/maintain.rs

```diff
@@ -571,70 +573,76 @@ fn changed_accounts_iter(
 
 async fn apply_local_txs_backup<P>(
     pool: P,
-    transactions_path: Option<PathBuf>,
-) -> Result<(), FsPathError>
+    file_path: &PathBuf,
+) -> Result<(), ApplyLocalTxsBackupError>
 where
     P: TransactionPool + TransactionPoolExt + 'static,
 {
-    if let Some(file_path) = transactions_path.clone() {
-        trace!(target: "reth::cli", txs_file =?file_path, "Check local persistent storage for saved transactions");
-        match reth_primitives::fs::read(&file_path) {
-            Ok(data) => {
-                warn!(target: "reth::cli", txs_file = ?file_path, "We read provided local transactions file and will try to decode transactions");
-                let mut data_slice = data.as_slice();
-                let txs_signed: Vec<TransactionSigned> =
-                    alloy_rlp::Decodable::decode(&mut data_slice).unwrap();
-
-                let pool_transactions = txs_signed
-                    .into_iter()
-                    .map(|tx| {
-                        let tx = tx.try_ecrecovered().unwrap();
-                        <P::Transaction>::from_recovered_transaction(tx)
-                    })
-                    .collect::<Vec<_>>();
-                let _ =
-                    pool.add_transactions(crate::TransactionOrigin::Local, pool_transactions).await;
-                Ok(())
-            }
-            Err(err) => {
-                error!(target: "reth::cli", 
-                    txs_file = ?file_path, 
-                    "Unable to read provided local transactions file. Encounetered error: {}", err);
-                Err(err)
-            }
-        }
-    } else {
-        Ok(())
-    }
+    info!(target: "reth::cli", txs_file =?file_path, "Check local persistent storage for saved transactions");
+    let data = reth_primitives::fs::read(file_path)?;
+    warn!(target: "reth::cli", txs_file = ?file_path, "We read provided local transactions file and will try to decode transactions");
+    let mut data_slice = data.as_slice();
+    let txs_signed: Vec<TransactionSigned> = alloy_rlp::Decodable::decode(&mut data_slice)?;
+
+    let pool_transactions = txs_signed
+        .into_iter()
+        .map(|tx| {
+            let tx = tx.try_ecrecovered().unwrap();
```

Comment:
this should not unwrap but instead use filter_map and simply skip failed recovered txs